### ISSUE TITLE
Ignore ENV['no_proxy'] while initializing the connection

### ISSUE THREAD

**Issue by cosmtrek** [lostisland/faraday#487]
Consider this scenario:  When ENV['http_proxy'] and ENV['no_proxy'] are set in my zsh config file, faraday always fires a request via proxy even this request should not go with proxy. 

Specifically, I have set ENV variables as follows:

```
# zsh.rc
export http_proxy=http://127.0.0.1:10240/
export HTTPS_PROXY=$http_proxy
export NO_PROXY='127.0.0.1, localhost'
```

If I get the request http://localhost:3000, faraday should not go with proxy.

I read faraday's source code and hack it as follows:

```
# lib/faraday/connection.rb line 83
proxy(options.fetch(:proxy) {
  uri = ENV['http_proxy']
  if uri && !uri.empty?
    uri = 'http://' + uri if uri !~ /^http/i
    no_proxy = ENV['no_proxy'].gsub(/\s+/, '').split(',')
    u = URI(uri)
    no_proxy.include?(u.hostname) ? nil : uri
  end
})
```

**Comment by raphox** [comment]
Duplicate https://github.com/lostisland/faraday/pull/471

**Comment by iMacTia** [comment]
This is being addressed in #637
